File: solutions/haskell/acronym/1/src/Acronym.hs
module Acronym (abbreviate) where
import Data.List
import Data.Char

abbreviate :: String -> String
abbreviate xs = intercalate "" (map (\w -> if all isUpper w then take 1 w; else [ c | c <- w, c `elem` ['A'..'Z']]) (map (\w -> toUpper (head w) : tail w) (words (map (\c -> if c=='\'' then 's'; else c) (map (\c -> if c=='-' || c=='_' || c==',' then ' '; else c) xs)))))

File: solutions/python/acronym/1/acronym.py
def abbreviate(words):
    words=words.replace('-',' ')
    words=words.replace('_','')
    word_list=words.split()
    print(word_list)
    acronym=""
    for x in word_list:
        acronym=acronym+x[0].upper()
    return acronym
